kolibri/content/utils/sqlalchemybridge.py

```diff
@@ -123,22 +125,43 @@ def prepare_bases():
 
         with open(SCHEMA_PATH_TEMPLATE.format(name=name), 'rb') as f:
             metadata = pickle.load(f)
-        BASES[name] = prepare_base(metadata)
+        cascade_relationships = name == CONTENT_SCHEMA_VERSION
+        BASES[name] = prepare_base(metadata, cascade_relationships=cascade_relationships)
 
 
-def prepare_base(metadata):
+def get_model_from_cls(cls):
+    return next((m for m in apps.get_models(include_auto_created=True) if m._meta.db_table == cls.__table__.name), None)
+
+
+def get_field_from_model_by_column(model, column):
+    return next((f for f in model._meta.fields if f.column == column), None)
+
+
+def prepare_base(metadata, cascade_relationships=False):
     """
     Create a Base mapping for models for a particular schema version of the content app
     A Base mapping defines the mapping from database tables to the SQLAlchemy ORM and is
     our main entrypoint for interacting with content databases and the content app tables
     of the default database.
+    If cascade_relationships is True, then also attempt to use Django model information
+    to setup proper relationship cascade behaviour to allow deletion in SQLAlchemy.
     """
     # Set up the base mapping using the automap_base method, using the metadata passed in
     Base = automap_base(metadata=metadata)
-    # TODO map relationship backreferences using the django names
     # Calling Base.prepare() means that Base now has SQLALchemy ORM classes corresponding to
     # every database table that we need
-    Base.prepare()
+
+    def _gen_relationship(base, direction, return_fn,
+                          attrname, local_cls, referred_cls, **kw):
+        if direction is interfaces.ONETOMANY:
+            kw['cascade'] = 'all, delete-orphan'
+        # Add cascade behaviour on deletion
+        return generate_relationship(base, direction, return_fn,
```

Comment:
@rtibbles Just an observation: you wrapped line #154 even though one-lining it would be shorter than e.g. #133. I guess this might be a good case for https://github.com/learningequality/kolibri/issues/3609.

Reply:
I think that was probably because I copied it initially from a code sample that was pep8 compliant.